### PR TITLE
[test] Skip tests for column pinning and dynamic row height

### PR DESCRIPTION
**Pull request by m4theushw**
#5782 added new tests for dynamic row height which don't pass on Chrome non-headless and Edge, but pass on Chrome headless and Safari. These tests are very flacky so it was better to skip them on certain browsers. We already had this problem in the past, e.g. #5707

### CODE REVIEW

File: packages/grid/x-data-grid-pro/src/tests/columnPinning.DataGridPro.test.tsx [hunk 1]
```diff
@@ -264,12 +264,16 @@ describe('<DataGridPro /> - Column pinning', () => {
...
 
   describe('dynamic row height', () => {
-    it('should work with dynamic row height', async function test() {
-      if (isJSDOM) {
-        // Need layouting
+    beforeEach(function beforeEach() {
+      const { userAgent } = window.navigator;
+
+      if (isJSDOM || (!userAgent.includes('Headless') && /safari/i.test(userAgent))) {
+        // Need layouting and on Chrome non-headless and Edge these tests are flacky
```
Comment (LukasTy) [inline packages/grid/x-data-grid-pro/src/tests/columnPinning.DataGridPro.test.tsx:271]: Is it `Edge` or `Safari`? 🤔

Comment (m4theushw) [reply packages/grid/x-data-grid-pro/src/tests/columnPinning.DataGridPro.test.tsx:271]: I updated the comment. It's to skip if Edge.

**Comment by DanailH** [comment]
I guess this means that PR is not needed https://github.com/mui/mui-x/pull/6075 ?

**Review by DanailH — APPROVED** [review]

**Comment by oliviertassinari** [comment]
Do we know why are these tests failing on Chrome and Edge? Could this hide a real problem with our implementation?

**Comment by m4theushw** [comment]
@oliviertassinari @cherniavskii once discovered that these tests only fail on Chrome non-headless. The implementation seems correct because I can't see any problem with dynamic row height when running on Edge. We also never had an issue with this specific browser.

**Comment by cherniavskii** [comment]
> @oliviertassinari @cherniavskii once discovered that these tests only fail on Chrome non-headless. The implementation seems correct because I can't see any problem with dynamic row height when running on Edge. We also never had an issue with this specific browser.

For the reference: https://github.com/mui/mui-x/issues/5095#issuecomment-1144720089

**Pull request merged**